### PR TITLE
Upgrade eslint to incuna's shared. Remove jscs

### PR DESCRIPTION
**Pull request by henrahmagix**
@incuna/js Please merge, ta!

It's a big diff, but here are the changes:
- `'use strict'` is moved to outside `function`
- trailing commas added
- `angular.noop` replaced empty functions `function () {}`
- snake_case object properties no longer need linting disabled

Thanks to [js-beautify](https://www.npmjs.com/package/js-beautify), used with git-diff [like this](https://cweiske.de/tagebuch/git-diff-minified-js.htm) (npm installed globally instead of pip), the dist diff is as such:
```diff
diff --git a/dist/angular-token-auth.min.js b/dist/angular-token-auth.min.js
index dfdb8fb..7dda3fa 100644
--- a/dist/angular-token-auth.min.js
+++ b/dist/angular-token-auth.min.js
@@ -24,7 +24,7 @@ function(a) {
 function(a) {
     "use strict";
     var b = a.module("angular-token-auth.auth-login", []);
-    b.controller("LoginCtrl", [function() {}])
+    b.controller("LoginCtrl", [a.noop])
 }(window.angular),
 function(a) {
     "use strict";
@@ -225,9 +225,9 @@ function(a) {
         var e = d.COOKIE_PATH ? d.COOKIE_PATH : c.baseHref(),
             f = {
                 noSupport: {
-                    set: function() {},
-                    get: function() {},
-                    clear: function() {}
+                    set: a.noop,
+                    get: a.noop,
+                    clear: a.noop
                 },
                 cookie: {
                     test: function() {
```

So the functional changes to this library are minimal.

### CODE REVIEW

**Comment by henrahmagix** [comment]
The Incuna rule for `object-curly-newline` was changed to stop trying to fix single-property nested objects onto one line. When eslint v4 is released, we can use the `consistent: true` option to allow nested objects with less than 3 properties, whilst enforcing nesting for >3 properties. See https://github.com/eslint/eslint/commit/b337738f50bdd2ea732172de3ed7036fc1879c83

**Comment by henrahmagix** [comment]
> Do we need a changelog update

@Minglee01 Quite right! Done in 061f98d

**Comment by henrahmagix** [comment]
Thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# Upcoming
+
+* Empty functions `function () {}` replaced by `angular.noop`
```
Comment (wytrych) [inline CHANGELOG.md]: You didn't mention that `jscs` was removed and `eslintrc` updated

Comment (henrahmagix) [inline CHANGELOG.md]: @wytrych Those don't affect the distribution so they're not required to be listed in the changelog; I don't think they're useful to read. Users of this package don't need to know about internal changes that don't affect the code they're using.

If they want to see those changes, the commit history will show them =)

**Pull request merged**